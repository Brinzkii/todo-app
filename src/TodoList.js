import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import uuid from 'react-uuid';

const TodoList = () => {
	const [todos, setTodos] = useState([]);

	const addTodo = (todo) => {
		const newTodo = {
			...todo,
			id: uuid(),
		};

		setTodos((todos) => [...todos, newTodo]);
	};

	const removeTodo = (id) => {
		const idx = todos.findIndex((todo) => todo.id === id);
		const todosCopy = [...todos];
		todosCopy.splice(idx, 1);
		setTodos([...todosCopy]);
	};

	return (
		<>
			<NewTodoForm addTodo={addTodo} />

			<ul className="TodoList">
				{todos.map((t) => (
					<li key={uuid()}>
						<Todo todo={t.todo} id={t.id} removeTodo={removeTodo} key={uuid()} />
					</li>
				))}
			</ul>
		</>
	);
};

export default TodoList;
