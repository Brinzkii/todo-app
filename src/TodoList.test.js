import { fireEvent, render } from '@testing-library/react';
import TodoList from './TodoList';

it('renders without crashing', () => {
	render(<TodoList />);
});

it('matches snapshot', () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it('can add a new todo', () => {
	const { getByLabelText, queryByTitle, queryByText } = render(<TodoList />);
	expect(queryByTitle('Todo-item')).not.toBeInTheDocument();

	const todoInput = getByLabelText('Todo:');
	const submitBtn = queryByText('Add Todo');

	fireEvent.change(todoInput, { target: { value: 'mow the grass' } });
	fireEvent.click(submitBtn);

	expect(queryByTitle('Todo-item')).toBeInTheDocument();
});

it('can remove a todo', () => {
	const { getByLabelText, queryByTitle, queryByText } = render(<TodoList />);
	const todoInput = getByLabelText('Todo:');
	const submitBtn = queryByText('Add Todo');

	fireEvent.change(todoInput, { target: { value: 'mow the grass' } });
	fireEvent.click(submitBtn);

	const removeBtn = queryByText('X');

	fireEvent.click(removeBtn);

	expect(queryByTitle('Todo-item')).not.toBeInTheDocument();
});
