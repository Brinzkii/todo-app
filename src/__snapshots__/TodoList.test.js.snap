// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Todo-form"
  >
    <label
      class="Todo-form-label"
      for="todo"
    >
      Todo:
    </label>
    <input
      class="Todo-form-input"
      id="todo"
      name="todo"
      placeholder="Cut the grass"
      type="text"
      value=""
    />
    <button
      class="Todo-form-button"
    >
      Add Todo
    </button>
  </form>
  <ul
    class="TodoList"
  />
</DocumentFragment>
`;
